string node_from
string node_to
